Adds Create, Delete, Retention (#2020)
temporalio/documentation: docs-src/cloud/tcld/namespace/retention/set.md

---
id: set
title: tcld namespace retention set
sidebar_label: set

> description: How to set the length of time a closed Workflows will be preserved before deletion for a Namespace in Temporal Cloud using tcld.

Reviewer: ```suggestion
description: How to set the length of time a closed Workflow will be preserved before deletion for a Namespace in Temporal Cloud using tcld.
```

tags:
  - tcld
---

> #### `set`
> 

Reviewer: ```suggestion
```

Set the length of time (in days) a closed Workflow will be preserved before deletion for the specified Namespace.

Alias: `s`

> The following modifier controls the behavior of the command.

Reviewer: ```suggestion
The following modifiers control the behavior of the command.
```

#### `--namespace`

_Required modifier_

> Specify the Namespace hosted on Temporal Cloud to be set.

Reviewer: ```suggestion
Specify a Namespace hosted on Temporal Cloud.
If not specified, the value of the environment variable $TEMPORAL_CLOUD_NAMESPACE is used.
```

Alias: `-n`

#### `--retention-days`

_Required modifier_

> Specify the length of time (in days) a closed Workflow will be preserved before deletion for the specified Namespace.

Reviewer: ```suggestion
Specify the number of days a closed Workflow will be preserved before deletion.
```
